support-frontend/app/controllers/RedemptionController.scala

```diff
@@ -50,26 +57,46 @@ class RedemptionController(
   val js = "subscriptionsRedemptionPage.js"
   val css = "digitalSubscriptionCheckoutPage.css" //TODO: Don't need this?
 
-  def displayForm(redemptionCode: RawRedemptionCode): Action[AnyContent] = googleAuthAction.async {
+  val getCorporateCustomer = new GetCorporateCustomer(dynamoLookup)
+  val testUserFromRequest = new TestUserFromRequest(testUsers)
+
+  def displayForm(redemptionCode: RawRedemptionCode): Action[AnyContent] = (googleAuthAction andThen maybeAuthenticatedAction()).async {
     implicit request =>
-      getCorporateCustomer(redemptionCode).fold(Some.apply, _ => None).map(
-        maybeError =>
-          Ok(subscriptionRedemptionForm(
-            title = title,
-            mainElement = id,
-            js = js,
-            css = css,
-            fontLoaderBundle = fontLoaderBundle,
-            csrf = None,
-            uatMode = false,
-            stage = "checkout",
-            redemptionCode = redemptionCode,
-            maybeRedemptionError = maybeError,
-            user = None,
-            submitted = false
-          )).withHeaders(CacheControl.noCache)
-      )
-    }
+      for {
+        whyIsTestUser <- request.user.map { authenticatedIdUser =>
```

Comment:
Could you break out the code involved in this assignment to a new method please? Also shouldn't it just be `isTestUser` rather than `whyIsTestUser`?

Reply:
I wasn't sure about naming, actually this also gives the reason they are a test user, e.g. whether it came from the cookie or the identity display name.

Reply:
how about `testUserStatus`?

Reply:
I'd go for `isTestUser` it is pretty standard in the code base both in the scala and js code

Reply:
Do we need it to be an `Option[String]` or can we map it to a boolean?

Reply:
ok will do, originally I specifically wanted to avoid that to make it clear this wasn't the usual true/false, but the IDE should keep people right via the type checkers anyway.

Reply:
to answer "need", the string on the true side explains whether you were marked as a test user due to the cookie or due to being logged in.  If that information is not useful then a boolean would be enough
![image](https://user-images.githubusercontent.com/7304387/85436512-1838b680-b581-11ea-8257-7d8c3d60248f.png)

Reply:
If the string is only so we can display the username in the flash I'd just turn this into a boolean and reword the flash so it doesn't use the displayname

Reply:
ok something like this: 
![image](https://user-images.githubusercontent.com/7304387/85438490-fb51b280-b583-11ea-9d00-dc6234d2a1dd.png)